src/components/EventDialog.js

```diff
@@ -0,0 +1,113 @@
+import React, { useCallback } from 'react';
+
+import Dialog from '@material-ui/core/Dialog';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import DialogContent from '@material-ui/core/DialogContent';
+import gql from 'graphql-tag';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/react-hooks';
+import { Grid, Button, Box } from '@material-ui/core';
+import useStyle from './UseStyle';
+
+const mutation = gql`
+  mutation deleteEvent($id: ID!) {
+    deleteEvent(id: $id) {
+      id
+    }
+  }
+`;
+
+function EventDialog(props) {
+  const { onClose, open, selectedEvent } = props;
+  const [deleteEvent, { loading, error }] = useMutation(mutation);
+
+  const classes = useStyle();
+
+  const handleClick = useCallback(() => {
+    deleteEvent({
+      variables: { id: selectedEvent ? selectedEvent.event.id : '' },
+    });
+    onClose();
+  }, [deleteEvent, selectedEvent, onClose]);
+  if (loading) return <div>Loading...</div>;
```

Comment:
nit: These 2 lines repeating in other files, maybe there is a better place to write it and import it.

Reply:
Thanks.
I want to merge it. so maybe in the next commit